openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -44,6 +44,19 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
       val docs: DocumentedJsonSchema
   )
 
+  class RecordReference[A](
+      val name: String,
+      referencedRecord: Record[A]
+  ) extends Record[A](referencedRecord.ujsonSchema, referencedRecord.docs.withName(name)) {
```

Comment:
Interesting - I was expecting to see a subclass like this on `DocumentedRecord`, not `Record`.

Should it ever be possible to have a `RecordReference` inside of a `RecordReference`? Eg. `RecordReference("foo", RecordReference("bar", RecordReference("baz", ...)))`? I see some effort here to avoid that case, but I can't tell yet if that is an invariant.